CONTRIBUTING.md

```diff
@@ -1,57 +1,95 @@
+   * good title: "Private registry push fail: 400 error with E_INVALID_DIGEST"
+ - [ ] copy the output of `docker version`
+ - [ ] copy the output of `docker info`
+ - [ ] copy your Registry version
+ - [ ] copy the command line you used to launch your Registry
+ - [ ] restart your docker daemon in debug mode (add `-D` to the daemon launch arguments)
+ - [ ] reproduce your problem and get your docker daemon logs showing the error
+ - [ ] if relevant, copy your registry logs that show the error
+ - [ ] provide any relevant detail about your specific Registry configuration (eg: storage backend used)
+ - [ ] indicate if you are using an enterprise proxy, Nginx, or anything else between you and your Registry
 
-Create a new proposal in the folder `open-design/specs`, named `DEP_MY_AWESOME_PROPOSAL.md`, using `open-design/specs/TEMPLATE.md` as a starting point.
+## Contributing a patch for a known bug, or a small correction
 
-Then immediately submit this new file as a pull-request, in order to get early feedback.
+You should follow the basic GitHub workflow:
 
-Eventually, you will have to update your proposal to accommodate the feedback you received.
+ - [ ] fork
+ - [ ] commit a change
+ - [ ] make sure the tests pass
+ - [ ] PR
 
-Usually, it's not advisable to start working too much on the implementation itself before the proposal receives sufficient feedback, since it can be significantly altered (or rejected).
+Additionally, you must [sign your commits](https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work). It's very simple:
+
+ * configure your name with git: `git config user.name Batman && git config user.email batman@batcave.com`
+ * sign your commits using `-s`: `git commit -s -m "My commit"`
+
+Some simple rules to ensure quick merge:
+
+ - [ ] clearly point to the issue(s) you want to fix in your PR comment
+ - [ ] when possible, prefer multiple (smaller) PRs addressing individual issues over a big one trying to address multiple issues at once
+ - [ ] if you need to amend your PR following comments, please squash instead of adding more commits
+
+## Contributing new features
+
+You are heavily encouraged to first discuss what you want to do. You can do so on the irc channel, or by opening an issue that clearly describes the use case you want to fulfill, or the problem you are trying to solve.
+
+If this is a major new feature, you should then submit a proposal that describes your technical solution and reasonning.
+If you did discuss it first, this will likely be greenlighted very fast. It's advisable to address all feedback on this proposal before starting actual work. 
+
+Then you should submit your implementation, clearly linking to the issue (and possible proposal).
+
+Your PR will be reviewed by the community, then ultimately by the project maintainers, before being merged.
+
+It's mandatory to:
+
+ * interact respectfully with other community members and maintainers
```

Comment:
You could include a link to the community guidelines https://github.com/docker/docker/blob/master/CONTRIBUTING.md#docker-community-guidelines

(Or copy them to this readme)